//----------------------------------------------------------------------------
// Copyright 2021, Ed Keenan, all rights reserved.
//----------------------------------------------------------------------------

syntax = "proto3";

option optimize_for = LITE_RUNTIME;

message vboData_proto
{
	enum VBO_TARGET
	{
		ARRAY_BUFFER = 0;
		ELEMENT_ARRAY_BUFFER = 1;
	};

	enum VBO_COMPONENT
	{
		BYTE = 0;
		UNSIGNED_BYTE = 1;
		SHORT = 2;
		UNSIGNED_SHORT = 3;
		INT = 4;
		UNSIGNED_INT = 5;
		FLOAT = 6;
		DOUBLE = 7;
	};

	enum VBO_TYPE
	{
		VEC2 = 0;
		VEC3 = 1;
		VEC4 = 2;
		MAT2 = 3;
		MAT3 = 4;
		MAT4 = 5;
		SCALAR = 6;
		VECTOR = 7;
		MATRIX = 8;
	};

	optional uint32			enabled		  = 1;
	optional VBO_TARGET     targetType    = 2;
	optional VBO_COMPONENT  componentType = 3;
	optional VBO_TYPE       vboType       = 4;
	optional uint32         attribIndex   = 5;
	optional uint32         count         = 6;
	optional uint32         dataSize      = 7;
	optional bytes          poData        = 8;

}

// -- End of File ---
